packages/gatsby/src/cache-dir/loader.js

```diff
@@ -67,7 +68,7 @@ const fetchResource = (resourceName, cb = () => {}) => {
     // Find resource
     const resourceFunction =
       resourceName.slice(0, 6) === `page-c`
-        ? asyncRequires.components[resourceName]
+        ? asyncRequires.components[resourceName] || asyncRequires.layouts[resourceName]
```

Comment:
Yea, it's created from their filename so we should be good using the same naming func.